src/main/java/io/lettuce/core/ClientOptions.java

```diff
@@ -63,6 +73,10 @@
 
     private final boolean autoReconnect;
 
+    private final boolean periodicReauthenticate;
```

Comment:
minor: can be removed, since `reauthenticatePeriod` is nullable. You can use the fact that `reauthenticatePeriod` was set as an indicator of whether periodic reauthentication should happen, and that will prevent bugs where only one of the values was set.

Reply:
minor: If you keep it, then rename to shouldReauthenticatePeriodically